http_server/main_test.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http/httptest"
+	"net/http"
+	"testing"
+	"strings"
+
+)
+
+func TestWelcome(t *testing.T) {
+	
+	data := &serviceData{}
+	req := httptest.NewRequest("GET", "/welcome/?name=mara", nil)
+
+	w := httptest.NewRecorder()
+	mux := data.muxSetup()
+	mux.ServeHTTP(w, req)
+
+	resp := w.Result()
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	if !strings.Contains(string(body), "mara") {
+		t.Error("Query parameters have not been parsed properly")
```

Comment:
In Go tests we're always trying to give a somewhat meaningful error message. In this case, "have not been parsed properly" doesn't say much. Tell exactly what you've been looking for, e.g. "didn't find 'mara' in response body".

That being said, I'd actually make the handler itself check properties and report them back (this is what we've discussed during today's sync).